classes/mock/generator.php

```diff
@@ -490,11 +490,39 @@ protected function generateMethodSignature(\reflectionMethod $method)
 
     protected function generateClassCode(\reflectionClass $class, $mockNamespace, $mockClass)
     {
+        $propertiesCode = '';
+
+        // PHP 8.0+ : Generate promoted properties from constructor
+        if (method_exists(\ReflectionParameter::class, 'isPromoted')) {
```

Comment:
Je pense que c'est préférable de se baser sur la version de PHP. En effet, si la méthode disparaît en PHP 9.x, ça génèrera une exception qui sera plus facilement détectable qu'un changement dans le mock généré qui ne sera détectable que si le test correspondant a prévu ce cas.

Aussi, s'il y a une erreur dans le nom de la méthode, le code échouera silencieusement et ça ne sera pas détecté si le test ne prévoie pas ce cas. C'est d'ailleurs probablement ce qui se pass plus bas avec `ReflectionProperty::isPublicSet()` qui d'après la doc PHP n'existe pas.

```suggestion
        if (version_compare(PHP_VERSION, '8.0.0', '>=')) {
```